test/model/modelInfo/ElementFactory.test.ts

```diff
@@ -0,0 +1,215 @@
+import ElementFactory, {
+  ElementXml,
+  ElementTypeSpecifierXml,
+} from "../../../src/model/modelInfo/ElementFactory";
+import ChoiceElement from "../../../src/model/modelInfo/ChoiceElement";
+import ListElement from "../../../src/model/modelInfo/ListElement";
+import SimpleElement from "../../../src/model/modelInfo/SimpleElement";
+
+describe("ElementFactory", () => {
+  let choiceSpecifier: ElementTypeSpecifierXml;
+  let listSpecifier: ElementTypeSpecifierXml;
+  let alternateListSpecifier: ElementTypeSpecifierXml;
+
+  beforeEach(() => {
+    choiceSpecifier = {
+      $: {
+        "xsi:type": "ns4:ChoiceTypeSpecifier",
+      },
+      "ns4:choice": [
+        {
+          $: {
+            modelName: "FHIR1",
+            name: "Name1",
+          },
+        },
+        {
+          $: {
+            modelName: "FHIR2",
+            name: "Name2",
+          },
+        },
+      ],
+    };
+
+    listSpecifier = {
+      $: {
+        "xsi:type": "ns4:ListTypeSpecifier",
+        elementType: "FHIR.TheType",
+      },
+    };
+
+    // This is the one with the NamedTypeSpecifier child
+    alternateListSpecifier = {
+      $: {
+        "xsi:type": "ns4:ListTypeSpecifier",
+      },
+      "ns4:elementTypeSpecifier": [
+        {
+          $: {
+            modelName: "FHIR3",
+            name: "Name3",
+          },
+        },
+      ],
+    };
+  });
+
+  describe("createElement()", () => {
+    it("should create a SimpleElement", () => {
+      const input: ElementXml = {
+        $: {
+          name: "memberName",
+          elementType: "namespace.typeName",
+        },
+      };
+      const result = ElementFactory.createElement(input);
+      expect(result).toBeInstanceOf(SimpleElement);
+      const castResult = result as SimpleElement;
+      expect(castResult.name).toBe("memberName");
+      expect(castResult.namespace).toBe("namespace");
+      expect(castResult.typeName).toBe("typeName");
+    });
+
+    it("should create a ListElement", () => {
+      const input: ElementXml = {
+        $: {
+          name: "listName",
+        },
+        "ns4:elementTypeSpecifier": [listSpecifier],
+      };
+      const result = ElementFactory.createElement(input);
+      expect(result).toBeInstanceOf(ListElement);
+      const castResult = result as ListElement;
+      expect(castResult.name).toBe("listName");
+      expect(castResult.namespace).toBe("FHIR");
+      expect(castResult.typeName).toBe("TheType");
+    });
+
+    it("should create a ListElement using the alternate syntax", () => {
+      const input: ElementXml = {
+        $: {
+          name: "alternateListName",
+        },
+        "ns4:elementTypeSpecifier": [alternateListSpecifier],
+      };
+      const result = ElementFactory.createElement(input);
+      expect(result).toBeInstanceOf(ListElement);
+      const castResult = result as ListElement;
+      expect(castResult.name).toBe("alternateListName");
+      expect(castResult.namespace).toBe("FHIR3");
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5966907)